lang/cs/Org.Apache.REEF.Client/Common/JobSubmissionResult.cs

```diff
@@ -96,6 +96,36 @@ public string GetUrlResult(string url)
             return task.Result;
         }
 
+        public void WaitForDriverToFinish()
+        {
+            DriverStatus status = FetchDriverStatus();
+            
+            while (status.IsActive())
+            {
+                try
+                {
+                    status = FetchDriverStatus();
+                }
+                catch (System.Net.WebException)
+                {
+                    // If we no longer can reach the Driver, it must have exited.
+                    status = DriverStatus.UNKNOWN_EXITED;
+                }
+            }
+        }
+
+        private DriverStatus FetchDriverStatus()
```

Comment:
May be we should also expose this API as public because client may want to know the driver current status. If it is only for knowing driver complete, we already have jobSubmitionResult.FinalState; that tell client the driver final state.

Reply:
As I said in the commit message, I don't want to do that just yet. A proper status API would be very useful, but needs more design considerations. For now, I want to unblock REEF-1271 without introducing any `public` APIs not needed for that goal.